platforms/core-execution/workers/src/integTest/groovy/org/gradle/workers/internal/WorkerExecutorProblemsApiIntegrationTest.groovy

```diff
@@ -163,39 +163,33 @@ class WorkerExecutorProblemsApiIntegrationTest extends AbstractIntegrationSpec {
             operationId == Long.parseLong(buildOperationIdFile.text)
             exception.message == "Exception message"
             exception.stacktrace.contains("Caused by: java.lang.Exception: Wrapped cause")
-            contextualLocations.size() == 1
-            (contextualLocations[0] as TaskLocation).buildTreePath == ":reportProblem"
-        }
-
-        def problem = Iterables.getOnlyElement(filteredProblemDetails(buildOperationsFixture))
```

Comment:
🤔 Why do we not check this here as well? It checks other methods on the progress event that will be relevant for the Develocity plugin.

Reply:
The `org.gradle.api.problems.internal.DefaultProblemProgressDetails#convertToLocation` filters the task location, so I can't check for it here.

the other check relies on `org.gradle.api.problems.internal.DefaultProblemProgressDetails#getProblem` to validate. I wasn't fully aware of this but we use it often.

But we could do both, if it makes sense to test that separately, or could we test that in a unit test?

because what effectively happens here is: `DefaultProblemProgressDetails` is serialized to json by using the methods not the actual values, so we would be testing the methods return values on the json.

Reply:
We want to test both. I added the test for the API that is used by the develocity plugin here since the setup is exactly the same I didn't see a use to split the test into two. We need to have the assertions for the Develocity API, I don't see a problem with asserting both parts.